src/compiler/typescript_to_rust/azle_generate/src/generators/header/timers.rs

```diff
@@ -0,0 +1,110 @@
+                        timer_callbacks_ref_cell.borrow_mut().remove(&timer_callback_id);
+                    });
+
+                    timer_callback_lookup.remove(&timer_id);
+                });
+            }
+
+            pub fn set_timer(
+                duration: Duration,
+                function: JsObject,
+                repeat: bool
+            ) -> TimerId {
+                let callback_id = uuid::Uuid::new_v4().to_string();
+
+                // We cannot pass the function directly to the closure because it's lifetime isn't
+                // long enough. It will go out of scope before it can be used by the closure. So
+                // instead, we store it in a global variable using a string key, and then later
+                // use that key (which can be passed to the closure) to look up the function.
+                TIMER_CALLBACKS_REF_CELL.with(|timer_callbacks_ref_cell| {
+                    let mut timer_callbacks = timer_callbacks_ref_cell.borrow_mut();
+
+                    timer_callbacks.insert(callback_id.clone(), TimerCallback {
+                        function,
+                        timer_id: TimerId::default() // This is just a placeholder until we create the timer below.
+                    })
+                });
+
+                let closure = create_callback_closure(callback_id.clone(), !repeat);
+
+                let timer_id = if repeat {
+                    ic_cdk::timer::set_timer_interval(duration, closure)
+                } else {
+                    ic_cdk::timer::set_timer(duration, closure)
+                };
+
+                TIMER_CALLBACKS_REF_CELL.with(|timer_callbacks_ref_cell|{
+                    let mut timer_callbacks = timer_callbacks_ref_cell.borrow_mut();
+
+                    timer_callbacks
+                        .entry(callback_id.clone())
+                        .and_modify(|timer_callback| timer_callback.timer_id = timer_id);
+                });
+
+                TIMER_CALLBACK_LOOKUP_REF_CELL.with(|timer_callback_lookup_ref_cell|{
+                    let mut timer_callback_lookup = timer_callback_lookup_ref_cell.borrow_mut();
+
+                    timer_callback_lookup.insert(timer_id, callback_id.clone());
+                });
+
+                timer_id
+            }
+
+            fn create_callback_closure(callback_id: String, should_auto_delete: bool) -> impl FnMut() + 'static {
+                move || {
+                    unsafe {
+                        let mut _azle_boa_context = crate::BOA_CONTEXT_OPTION.as_mut().unwrap();
+
+                        let timer_id = TIMER_CALLBACKS_REF_CELL.with(|timer_callbacks_ref_cell| {
+                            let timer_callbacks = timer_callbacks_ref_cell.borrow();
+
+                            let timer_callback = timer_callbacks.get(&callback_id).unwrap();
+
+                            crate::_azle_handle_boa_result(
```

Comment:
We need to discuss what to do here,  because it might panic, and we could leave dangling state